### PR TITLE
Move to 1ES pipeline templates for releases

### PR DESCRIPTION
**Pull request by idavis**

### CODE REVIEW

File: .ado/release.yml [hunk 1]
```diff
@@ -1,212 +1,269 @@
...
+    arch: x86_64
+    ARCHFLAGS: '-arch x86_64'
+  - name: mac_aarch64
+    poolName: 'Azure Pipelines'
+    imageName: 'macOS-latest'
+    os: macOS
+    arch: aarch64
+    ARCHFLAGS: '-arch arm64'
+    additionalRustTargets: aarch64-apple-darwin
+  - name: mac_universal
```
Comment (billti) [inline .ado/release.yml:56]: If we're doing mac universal, do we also need x64 and aarch64 specific runs?

File: .ado/release.yml [hunk 2]
```diff
@@ -1,212 +1,269 @@
...
-     FolderLocation: '$(System.DefaultWorkingDirectory)/target/wheels'
-     Owners: '$(OwnerPersonalAlias)@microsoft.com'  # NB: Group email here fails the task with non-actionable output.
-     Approvers: 'billti@microsoft.com'
-     # Auto-inserted Debugging defaults:
-     ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
-     MainPublisher: 'QuantumDevelpmentKit'          # ESRP Team's Correction (including the critical typo "Develpm").
-     DomainTenantId: '72f988bf-86f1-41af-91ab-2d7cd011db47'
+  LLVM_VERSION: 14
+
+resources:
```
Comment (billti) [inline .ado/release.yml:27]: Being this is a public repo in a different org, we should maybe have some comments explaining what these 1ES templates are.

File: .ado/release.yml [hunk 3]
```diff
@@ -1,212 +1,269 @@
...
+              Write-Host "##vso[task.setvariable variable=QIRLIB_CACHE_DIR;]$directory"
+              Write-Host "##vso[task.setvariable variable=QIRLIB_DOWNLOAD_LLVM;]false"
+            displayName: Configure Environment
+            condition: succeeded()
+
+          # prerequisites for building
+          - script: |
+              sudo apt-get install -y ninja-build
+            displayName: Install build dependencies
+            condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'), ne(variables.CACHE_RESTORED, 'true'))
```
Comment (billti) [inline .ado/release.yml:165]: Is all the caching stuff new too?

Comment (idavis) [reply .ado/release.yml:165]: Caching was used in the sdl pipeline but I hadn't yet brought it over to the release until now.

**Pull request merged**